packages/shell-api/src/bulk.ts

```diff
@@ -0,0 +1,217 @@
+import { hasAsyncChild, returnsPromise, ShellApiClass, shellApiClassDefault } from './decorators';
+import Mongo from './mongo';
+import { MongoshInternalError, MongoshInvalidInputError, MongoshUnimplementedError } from '@mongosh/errors';
+import {
+  Document,
+  WriteConcern
+} from '@mongosh/service-provider-core';
+import { assertArgsDefined } from './helpers';
+import { BulkWriteResult } from './result';
+
+@shellApiClassDefault
+export class BulkFindOp {
+  _innerFind: any;
```

Comment:
if _innerFind and _innerBulk are coming from service provider probably making that clear would help who reads to orientate a bit: `_serviceProviderBulkOp`, `_serviceProviderBulkOpFind`

Reply:
Can do!